File: Exemplos M10/App_IntentDigitalHUB_ReactNative/Elgin_IntentDigitalHUB_ReactNative-font/src/intentDigitalHubService/BALANCA/BalancaCommand.ts
import {IntentDigitalHubCommand} from '../IntentDigitalHubCommand';
import {IntentDigitalHubModule} from '../IntentDigitalHubModule';

/**
 * Classe que generaliza todos os comandos Balanca definindo o módulo do comando e o seu tipo de retorno
 */
export abstract class BalancaCommand extends IntentDigitalHubCommand {
  constructor(functionName: string) {
    super(functionName, IntentDigitalHubModule.BALANCA);
  }
}

File: Exemplos M10/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_Kotlin-font/app/src/main/java/com/elgin/intent_digital_hub/IntentDigitalHubService/BALANCA/BalancaCommand.kt
package com.elgin.intent_digital_hub.IntentDigitalHubService.BALANCA

import com.elgin.intent_digital_hub.IntentDigitalHubService.IntentDigitalHubCommand
import com.elgin.intent_digital_hub.IntentDigitalHubService.IntentDigitalHubModule
import com.google.gson.annotations.SerializedName
import java.io.Serializable


/**
 * Classe que generaliza todos os comandos do módulo BALANCA, definindo o módulo do comando e o seu tipo de retorno
 */

//O módulo balança possuí diferentes tipos de retorno para cada função, por isto é utilizado Generics para o serializável

abstract class BalancaCommand<T> protected constructor(functionName: String?) : IntentDigitalHubCommand(functionName, IntentDigitalHubModule.BALANCA), Serializable  {
    @SerializedName("resultado")
    val resultado : T? = null;
}